### DISCUSSION TITLE
The hub disconnected before flashing of the firmware was completed

### DISCUSSION THREAD

**Discussion by speelvogelke** [pybricks/support#792] [Q&A] [answered]
I wanted to start experimenting with the Pybricks Software/Firmware. But flashing the firmware gave problems.
Now I want to figure out what's going wrong.
Included is the capture of the bluetooth capture (wireshark) , because I think it has something to do with the ASUS BT500 adapter.

Describe the bug
Cannot install City Hub firmware in Microsoft Edge and Google Chrome on Windows.

What happens is the following :
- It pairs just fine, and appears to erase the flash as usual.
- The light is off for a bit, then starts blinking red/green/blue.
- But Pybricks Firmware doesn"t appear to start flashing. I receive an error

Pybricks Code v3.1.0
Chrome Version 107.0.5304.107 (Official build) (64-bits)
Edge Version 107.0.1418.42 (Official build) (64-bit)
Windows 10 
Bluetooth adapter used = ASUS BT500

Expected behavior
Doesn't start installation after erasing flash.
Error = The hub was disconnected before flashing was completed. Restart hub and try again.

What I tried already 
- restart the hub
- new batteries
- update ASUS driver manually
- using the beta Pybricks
- using another City Hub

Any help is welcome.
Thanks!

Dumpfile wireshark capture
[dump test.txt](https://github.com/pybricks/support/files/9992310/dump.test.txt)

**Comment by dlech** [comment, marked as answer]
Thanks for the report!

I'm not able to open the log in Wireshark. It looks like a text file with just some basic info. Can you zip up the full log and upload it here?

**Reply by speelvogelke** [reply]
The text file is a copy of the outputwindow , I include the wireshark file now ;-)

[BT500.zip](https://github.com/pybricks/support/files/9992573/BT500.zip)

**Reply by dlech** [reply]
Thanks for this.

![image](https://user-images.githubusercontent.com/963645/201408888-12838292-368c-43ec-8124-2c20208c9acc.png)

Here we can see that the request to get the device name is interleaved with erase flash command which could be causing problems for the LEGO bootloader.

We'll try adding a delay to allow the device name to be read before starting the erase.

**Reply by speelvogelke** [reply]
Thanks for checking the file.

If you need more info or want me to test something, just let me know !

**Reply by dlech** [reply]
I just published Pybricks Beta v2.0.0-beta.11 with the added delay if you want to try again and see if it makes a difference when flashing firmware on Windows.

**Reply by speelvogelke** [reply]
Thank you dlech.

Works perfectly now !

**Comment by laurensvalk** [comment]
If you haven't already, don't forget to unplug any sensors from port A.

If you have an Android phone, you could try it with that to verify that it isn't something with the hub.

**Reply by speelvogelke** [reply]
Thanks for the tip.
But I already disconnected everything such as motors/lights.

With an android phone I was able to do the flashing of the firmware without problems. Great !
But I was thinking why the desktop is not an option ? In my case it would be easier to program the hub with my desktop.
Or can I flash the firmware and still program the hub with the desktop ?

**Reply by dlech** [reply]
You should be able to use the desktop for programming after flashing the firmware with Android. 

For background, flashing the firmware requires using the LEGO bootloader on the hub which can't be updated and has some Bluetooth issues. But once the firmware is flashed, Bluetooth should work better since we are able to make changes to the Bluetooth code there.

**Reply by speelvogelke** [reply]
Thanks for the swift reply !

I was thinking the same, so I updated the Lego Firmware to the latest version as well. (Older version was on the hub)
But it was not the solution.

Anyway, I will update once more with the Android to be able to tackle the Bluetooth issue.
